### PR TITLE
tweak stack

### PR DESCRIPTION
**Pull request by Lampese**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new generic stack data structure with basic operations.
	- Added functions for enhanced stack manipulation and conversion.
- **Refactor**
	- Improved stack implementation for better functionality and clarity.
- **Documentation**
	- Updated stack documentation for improved readability and understanding.

### CODE REVIEW

**Comment by Lampese** [comment]
@CAIMEOX Could you help me write a new README?

**Comment by CAIMEOX** [comment]
> @CAIMEOX Could you help me write a new README?

Sure.

File: stack/stack.mbt [hunk 1]
```diff
@@ -12,150 +12,162 @@
...
 /// ```
-/// let s: Stack[Int] = Stack::from_list(Cons(1, Cons(2, Cons(3, Nil))))
+/// let s: Stack[Int] = Stack::[1, 2, 3]
 /// let s1: Stack[Int] = Stack::from_stack(s)
-/// println(s1) // output: {elements: Cons(3, Cons(2, Cons(1, Nil))), len: 3}
+/// println(s1) // output: Stack::[3, 2, 1]
 /// ```
 pub fn Stack::from_stack[T](other : Stack[T]) -> Stack[T] {
   { ..other,  }
 }
```
Comment (CAIMEOX) [inline stack/stack.mbt:98]: duplicated with `copy`?

**Comment by CAIMEOX** [comment]
The `to_string` function seems counterintuitive which outputs the reversed element order of constructors with prefix `from` 
```moonbit
let st = Stack::[1, 2, 3]
st.to_string() // Stack::[3, 2, 1]
```

**Comment by Lampese** [comment]
> The `to_string` function seems counterintuitive which outputs the reversed element order of constructors with prefix `from`
> 
> ```
> let st = Stack::[1, 2, 3]
> st.to_string() // Stack::[3, 2, 1]
> ```

I was actually thinking about this, and the author seems to have intended it to be designed that way, and I was hoping to get his opinion on it.

cc @muqiuhan

**Pull request merged**